### ISSUE TITLE
Not working correctly when used with function head without body

### ISSUE THREAD

**Issue by ludwikbukowski** [arjan/decorator#18]
In Elixir, there is a convention that we use empty function head, which we use when we want to document the function and to define default arguments. I think `decorator` does not work correctly with such an empty head.

Example:
```
use Decorator.Define, [myfun: 0]

  def myfun(body, _context) do
    quote do
      unquote(body)
    end
  end


  def myfun(test)

  @decorate myfun()
  def myfun(_test) do
    IO.puts "test"
  end
```
Will generate warning:
```
warning: this clause cannot match because a previous clause at line XX always matches
``` 
Is it possible that it creates new head with body, that just overmatch following heads?
I mean something like:
```
  def myfun(_test) do
    nil
  end
  def myfun(_test) do
    IO.puts "test"
  end
```

**Comment by ludwikbukowski** [comment]
https://github.com/arjan/decorator/pull/27

**Comment by arjan** [comment]
Fixes in #27
